examples/guides/file-integration/ftp-listener/README.md

```diff
@@ -0,0 +1,236 @@
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+        basicAuth: {
+            username: config:getAsString("FTP_USERNAME"),
+            password: config:getAsString("FTP_PASSWORD")
+        }
+    }
+};
+```
+
+Create a ftpClient object.
+
+```ballerina
+ftp:Client ftpClient = new(ftpConfig);
+```
+
+Create the file listening resource which will listen to a remote directory on the ftp server and periodically notify the addition of a file with the specified file pattern.
+
+```ballerina
+service monitor on remoteServer {
+    resource function fileResource(ftp:WatchEvent m) {
+        foreach ftp:FileInfo v1 in m.addedFiles {
+            var proRes = processFile(untaint v1.path);
+
+             if (proRes == MOVE) {
+                // implementation
+
+                string destFilePath = createFolderPath(v1,conf.destFolder);
+                error? renameErr = ftpClient->rename(v1.path, destFilePath);
+
+            } else if (proRes == DELETE) {
+                // implementation
+            } else {
+               // implementation
+            }
+        }
+    }
+```
+Implement the file processing logic inside the processFile() function.<br/>
+
+This function will return MOVE or DELETE of Operation type based on the functionality we defined in the configs if the file processing is successful or an ERROR if the processing fails.
+
+```ballerina
+public function processFile(string sourcePath) returns Operation {
+
+    var getResult = ftpClient->get(sourcePath);
+    Operation res = ERROR;
+
+    // implementation
+    // returns MOVE or DELETE if processing is successful
+    // returns ERROR if processing fails
+
+    return  res;
+}
+```
+
+We can have the following functions to generate the file paths where the file needs to be moved after processing or after failure in processing.
+
+```ballerina
+public function createFolderPath(ftp:FileInfo v2,string folderPath) returns string {
+    string p2 = createPath(v2);
+    string path = folderPath + "/" + p2;
+    return path;
+}
+
+public function createPath(ftp:FileInfo v3) returns string {
+    int subString = v3.path.lastIndexOf("/");
+    int length = v3.path.length();
+    string subPath = v3.path.substring((subString + 1), length);
+    return subPath;
+}
+```
+### Deployment
+#### Deploying Locally
+
+To deploy locally, navigate to *guides/file_integration*, and execute the following command.
+
+```
+$ ballerina build
+```
+
+This builds a Ballerina executable archive (.balx) of the services that you developed in the target folder.
+You can run them with the command:
+
+```
+$ ballerina run <Executable_File_Name>
+```
+
+#### Deploying on Docker
+
+If necessary you can run the service that you developed above as a Docker container. Ballerina language includes a Ballerina_Docker_Extension, which offers native support to run Ballerina programs on containers.
```

Comment:
We are going to create a guide for docker/ Kubernates. Maybe we should link to those resources when this is done. Shall we create a critical issue to track this?